src/libcore/tests/iter.rs

```diff
@@ -2280,6 +2280,32 @@ fn test_skip_try_folds() {
     assert_eq!(iter.next_back(), Some(24));
 }
 
+#[test]
+fn test_skip_nth_back() {
```

Comment:
I don't think the comment above was addressed; if disagree please add a test that demonstrates that the underlying iterator is still advanced.

(You could do that with inspect, or something like `let mut it = ...; it.by_ref().skip(10).nth_back(100); assert_eq!(it.next_back(), ...);`)

Reply:
Ok, I didn't understand you meant that.

Reply:
You're now calling `self.len()` three times, maybe it's better to store it in a variable first.